compiler/src/iree/compiler/Preprocessing/Common/ApplyPDLLPatterns.cpp

```diff
@@ -0,0 +1,139 @@
+// Copyright 2024 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "iree/compiler/Preprocessing/Common/Passes.h"
+#include "iree/compiler/Preprocessing/Common/Utilities.h"
+
+#include "iree/compiler/Codegen/Dialect/Codegen/IR/IREECodegenDialect.h"
+#include "iree/compiler/Codegen/Dialect/Codegen/IR/IREECodegenOps.h"
+#include "iree/compiler/Dialect/Flow/IR/FlowOps.h"
+#include "iree/compiler/Dialect/Stream/IR/StreamOps.h"
+#include "iree/compiler/Dialect/Util/IR/UtilDialect.h"
+#include "llvm/Support/SourceMgr.h"
+#include "mlir/Dialect/Arith/IR/Arith.h"
+#include "mlir/Dialect/Func/IR/FuncOps.h"
+#include "mlir/Dialect/MemRef/IR/MemRef.h"
+#include "mlir/Dialect/PDL/IR/PDL.h"
+#include "mlir/Dialect/PDLInterp/IR/PDLInterp.h"
+#include "mlir/Dialect/Tensor/IR/Tensor.h"
+#include "mlir/IR/PatternMatch.h"
+#include "mlir/Parser/Parser.h"
+#include "mlir/Support/FileUtilities.h"
+#include "mlir/Tools/PDLL/AST/Context.h"
+#include "mlir/Tools/PDLL/AST/Nodes.h"
+#include "mlir/Tools/PDLL/CodeGen/CPPGen.h"
+#include "mlir/Tools/PDLL/CodeGen/MLIRGen.h"
+#include "mlir/Tools/PDLL/ODS/Context.h"
+#include "mlir/Tools/PDLL/Parser/Parser.h"
+#include "mlir/Transforms/GreedyPatternRewriteDriver.h"
+
+#define DEBUG_TYPE "iree-preprocessing-apply-pdll-patterns"
+
+using namespace mlir;
+using namespace mlir::iree_compiler;
+
+namespace mlir::iree_compiler::Preprocessing {
+
+#define GEN_PASS_DEF_APPLYPDLLPATTERNSPASS
+#include "iree/compiler/Preprocessing/Common/Passes.h.inc" // IWYU pragma: export
+
+} // namespace mlir::iree_compiler::Preprocessing
+
+// Populate patterns from PDLL files.
+static LogicalResult
+populatePDLLModuleFromFileName(MLIRContext *context,
+                               RewritePatternSet &patterns,
+                               llvm::StringRef pdllModuleFileName) {
+// TODO(#17233)
+// Its easier to give these functions definitions than try to dissable
+// tblgen, just return failure
+#ifndef IREE_LLVM_BUNDLED
```

Comment:
This is not great, but I understand why this is done this way. I am OK to land this cause this is two experimental features (byo_llvm and pdll support) clashing. It would probably be better if this wasnt built at all if `IREE_LLVM_BUNDLED` wasnt set, but I dont know if there is an alternative. I am OK with this for the time being.

Reply:
I think there is a better solution for this. Change the following to enable the tablegen individually:
https://github.com/iree-org/iree/blob/b8c831b5c7611ae5122db9b5ff343ea2bd0f7e83/compiler/src/iree/compiler/Preprocessing/Common/Passes.cpp#L12

`GEN_PASS_REGISTRATION` doesn't do anything else but define the individual macros